### PR TITLE
feat:[#756] implement Ltrim and Rtrim function

### LINKED ISSUE #756: [AST Builder] `Ltrim` and `Rtrim` function
# Backgroud

The following issue is a follow-up issue of #614.

# Task

- [x] Implement `Ltrim` function
- [x] Implement `Rtrim` function

The following issue is to implement the  `Ltrim` and `Rtrim` function in the `ast_builder` module to support the  `Ltrim` and `Rtrim` function in the query builder.
The following test should pass in order to complete the following task.

```rust
   #[test]
    fn function_rtrim() {
        let actual = rtrim(text("GlueSQL      "),None);
        let expected = "RTRIM('GlueSQL      ')";
        test_expr(actual, expected);

        let actual = text("GlueSQL      ").rtrim(None);
        let expected = "RTRIM('GlueSQL      ')";
        test_expr(actual, expected);

        let actual = rtrim(text("GlueSQLABC"),Some(text("ABC")));
        let expected = "RTRIM('GlueSQLABC','ABC')";
        test_expr(actual, expected);

        let actual = text("GlueSQLABC").rtrim(Some(text("ABC")));
        let expected = "RTRIM('GlueSQLABC','ABC')";
        test_expr(actual, expected);
    }

    #[test]
    fn function_ltrim() {
        let actual = ltrim(text("      GlueSQL"),None);
        let expected = "LTRIM('      GlueSQL')";
        test_expr(actual, expected);

        let actual = text("      GlueSQL").ltrim(None);
        let expected = "LTRIM('      GlueSQL')";
        test_expr(actual, expected);

        let actual = ltrim(text("ABCGlueSQL"),Some(text("ABC")));
        let expected = "LTRIM('ABCGlueSQL','ABC')";
        test_expr(actual, expected);

        let actual = text("ABCGlueSQL").ltrim(Some(text("ABC")));
        let expected = "LTRIM('ABCGlueSQL','ABC')";
        test_expr(actual, expected);
    }
```

### PR DESCRIPTION
**Pull request by ChobobDev**
resolves #756 

I have implemented `Ltrim` and `Rtrim` functions to ast_builder.

Many thanks

### CODE REVIEW

File: core/src/ast_builder/expr/function.rs [hunk 1]
```diff
@@ -132,6 +134,18 @@ impl TryFrom<FunctionNode> for Function {
             FunctionNode::Concat(expr_list) => expr_list.try_into().map(Function::Concat),
             FunctionNode::Degrees(expr) => expr.try_into().map(Function::Degrees),
             FunctionNode::Radians(expr) => expr.try_into().map(Function::Radians),
+            FunctionNode::Ltrim(expr_node, chars_node) => {
+                let chars = chars_node.map(|chars| chars.try_into().unwrap());
```
Comment (ever0de) [inline core/src/ast_builder/expr/function.rs:138]:
```suggestion
                let chars = chars_node.map(TryInto::try_into).transpose()?;;
```

<https://doc.rust-lang.org/stable/core/option/enum.Option.html#method.transpose>

Comment (ChobobDev) [reply core/src/ast_builder/expr/function.rs:138]: Thank you, I will apply the following changes as guided.

File: core/src/ast_builder/expr/function.rs [hunk 2]
```diff
@@ -349,12 +369,26 @@ pub fn radians<V: Into<ExprNode>>(expr: V) -> ExprNode {
...
+    )))
+}
+
+pub fn rtrim<T: Into<ExprNode>>(expr: T, chars: Option<T>) -> ExprNode {
+    ExprNode::Function(Box::new(FunctionNode::Rtrim(
+        expr.into(),
+        chars.map(|chars_node| chars_node.try_into().unwrap()),
+    )))
+}
+
```
Comment (ever0de) [inline core/src/ast_builder/expr/function.rs:385]: You have to change it here too.

Comment (ChobobDev) [reply core/src/ast_builder/expr/function.rs:385]: Oh, thank you for pointing this out, i will make those changes ASAP

File: core/src/ast_builder/expr/function.rs [hunk 3]
```diff
@@ -145,6 +153,22 @@ impl TryFrom<FunctionNode> for Function {
                 let start = start.try_into()?;
                 Ok(Function::Substr { expr, start, count })
             }
+            FunctionNode::Ltrim {
+                expr_node,
+                chars_node,
+            } => {
+                let chars = chars_node.map(TryInto::try_into).transpose()?;
```
Comment (ChobobDev) [inline core/src/ast_builder/expr/function.rs:160]: While reading through the following file, I found the repetition of the following line of code.
I thought about making the `Optional<Exprnode>` as our new node. 
I would like to ask for your ideas @panarch @ever0de

File: core/src/ast_builder/expr/function.rs [hunk 4]
```diff
@@ -56,6 +56,14 @@ pub enum FunctionNode {
         start: ExprNode,
         count: Option<ExprNode>,
     },
+    Ltrim {
+        expr_node: ExprNode,
+        chars_node: Option<ExprNode>,
```
Comment (panarch) [inline core/src/ast_builder/expr/function.rs:61]: It would be enough with `Ltrim { expr, chars }` without `_node` prefixes.

**Review by panarch — APPROVED** [review]
Looks all good. Thanks a lot 👍 👍

**Pull request merged**